invite/src/invite_bot.ts

```diff
@@ -67,8 +93,76 @@ export class InviteBot {
    * Should be called in response to new comments with the /invite or /tryassign
    * macros.
    */
-  async tryInvite(invite: Invite): Promise<boolean> {
-    return false;
+  async tryInvite(invite: Invite): Promise<void> {
+    const existingInvites = await this.record.getInvites(invite.username);
+    const pendingInvite = !!existingInvites.length;
+
+    if (pendingInvite) {
+      return await this.handlePendingInvite(invite);
+    }
+
+    // The user has not been invited by the bot yet, so try and send an invite.
+    let invited = false;
+    try {
+      invited = await this.github.inviteUser(invite.username);
+    } catch (error) {
+      return await this.handleErrorSendingInvite(invite, error);
+    }
+
+    if (!invited) {
+      return await this.handleUserAlreadyMember(invite);
+    }
+   
+    await this.handleNewInviteSent(invite);
+  }
+
+  /** Handle case where there's at least one pending invite already recorded. */
+  private async handlePendingInvite(invite: Invite): Promise<void> {
+    await this.record.recordInvite(invite);
+    await this.github.addComment(
+      invite.repo,
+      invite.issue_number,
+      `You asked me to invite \`@${invite.username}\` to \`${this.org}\`, ` +
+        'but they already have an invitation pending! I will update this ' +
+        'thread when the invitation is accepted.'
+    );
+  }
+
+  /** Handle case where the user is already a member. */
+  private async handleUserAlreadyMember(invite: Invite): Promise<void> {
+    await this.github.addComment(
+      invite.repo,
+      invite.issue_number,
+      `You asked me to invite \`@${invite.username}\`, but they are ` +
+        `already a member of \`${this.org}\`!`
+    );
+  }
+
+  /** Handle case where the invite was actually sent. */
+  private async handleNewInviteSent(invite: Invite): Promise<void> {
+    await this.record.recordInvite(invite);
+    await this.github.addComment(
+      invite.repo,
+      invite.issue_number,
+      `An invitation to join \`${this.org}\` has been sent to ` +
+        `\`@${invite.username}\`. I will update this thread when the ` +
+        'invitation is accepted.'
+    );
+  }
+
+  /** Handle case where the there was an error sending the invite. */
+  private async handleErrorSendingInvite(invite: Invite, error: Error): Promise<void> {
+    this.logger.error(
+      `Failed to send an invite to \`@${invite.username}\`: ${error}`
+    );
+    await this.github.addComment(
+      invite.repo,
+      invite.issue_number,
+      `You asked me to send an invite to \`@${invite.username}\`, but I ` +
+        'ran into an error when I tried. Try sending the invite manually.'
```

Comment:
How about :%s/Try sending the invite manually./You can try again later ask @ampproject/wg-infra for help./

And because you want to make this repo-agnostic, you can make the helping tag name an env variable :)

Reply:
How about "You can try sending the invite manually, or ask `@ampproject/wg-infra` for help."

Reply:
Done; resolving.

Reply:
sg :P